### PR TITLE
feat(step-generation, shared-data): add waste chute commands to types & atomic commands

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-849 and RAUT-766

# Overview

This PR lays the ground work for wiring up the waste chute commands in Step-generation.

1. add `moveToAddressableArea` command type to schemaV8 commands, as well as `dispenseInPlace`
2. create atomic commands for `moveToAddressableArea`, `dispenseInPlace`, `dropTipInPlace`, and `blowoutInPlace`
3. creates a util that puts the array of commands together for waste chute: `wasteChuteCommandsUtil`

nothing is wired up yet!

# Test Plan

Nothing is wired up so just look through the types and the step-generation atomic commands and make sure things look right and the params match.

See here for params: https://opentrons.atlassian.net/wiki/spaces/RPDO/pages/3862200335/Deck+Configuration+Addressable+Areas+PE+Command+Shapes

# Changelog

- add run time command and create command types for the 2 commands
- create atomic commands ( `moveToAddressableArea`, `dispenseInPlace`, `dropTipInPlace`, and `blowoutInPlace`) and write tests
- create `wasteChuteCommandsUtil` util and write test
- create new error creator: `addtionalEquipmentDoesNotExist`
- add some ToDos for updating robot state

# Review requests

see test plan

# Risk assessment

low

### CODE REVIEW

File: shared-data/command/types/setup.ts [hunk 1]
```diff
@@ -39,6 +39,19 @@ export interface MoveLabwareRunTimeCommand
...
+  extends CommonCommandCreateInfo {
+  commandType: 'moveToAddressableArea'
+  params: MoveToAddressableAreaParams
+}
+export interface MoveToAddressableAreaRunTimeCommand
+  extends CommonCommandRunTimeInfo,
+    MoveToAddressableAreaCreateCommand {
+  //  TODO(jr, 11/3/23): add to result type
+  result?: {}
+}
```
Comment (jerader) [inline shared-data/command/types/setup.ts:53]: Do we know what the result type should be?

File: shared-data/command/types/gantry.ts [hunk 2]
```diff
@@ -77,23 +77,37 @@ export interface RetractAxisRunTimeCommand
...
+export interface MoveToAddressableAreaCreateCommand
+  extends CommonCommandCreateInfo {
+  commandType: 'moveToAddressableArea'
+  params: MoveToAddressableAreaParams
+}
+export interface MoveToAddressableAreaRunTimeCommand
+  extends CommonCommandRunTimeInfo,
+    MoveToAddressableAreaCreateCommand {
+  //  TODO(jr, 11/6/23): add to result type
+  result?: {}
```
Comment (jerader) [inline shared-data/command/types/gantry.ts:89]: Do we know what the result type should be?

Comment (b-cooper) [reply shared-data/command/types/gantry.ts:89]: I don't think we do yet, so a TODO is fine for now

File: step-generation/src/commandCreators/atomic/moveToAddressableArea.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  addressableAreaName: string
+}
+export const moveToAddressableArea: CommandCreator<MoveToAddressableAreaArgs> = (
+  args,
+  invariantContext,
+  prevRobotState
+) => {
+  const { pipetteId, addressableAreaName } = args
+
+  // No-op if there is no pipette
```
Comment (b-cooper) [inline step-generation/src/commandCreators/atomic/moveToAddressableArea.ts]: is this consistent with what we'd do elsewhere in SG?  It seems to me like we'd probably create an error?

File: step-generation/src/utils/wasteChuteCommandsUtil.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import { getHasWasteChute } from './misc'
+import { reduceCommandCreators } from './reduceCommandCreators'
+import { curryCommandCreator } from './curryCommandCreator'
+import type {
+  CommandCreator,
+  CommandCreatorError,
+  CurriedCommandCreator,
+} from '../types'
+
+export type WasteChuteCommandsTypes = 'dispense' | 'blow out' | 'drop tip'
```
Comment (b-cooper) [inline step-generation/src/utils/wasteChuteCommandsUtil.ts]: the spaces in this internal enumeration seem inconsistent with the rest of our codebase, is there are reason they're not camelCase?

**Review by b-cooper — APPROVED** [review]
A few nitpicks. Nothing that can't be addressed in a follow up 🍁

File: step-generation/src/utils/wasteChuteCommandsUtil.ts (hunk 4 above)
Comment (jerader) [reply step-generation/src/utils/wasteChuteCommandsUtil.ts]: it was so i could plug them into the error creator, but ya i should change these to camelCase then split up the strings for the error creator in another const.

**Pull request merged**